app/models/user.rb

```diff
@@ -32,4 +32,9 @@ def ranked?
   def games_needed_to_rank
     games_finished >= GAMES_NEEDED_TO_RANK ? 0 : GAMES_NEEDED_TO_RANK - games_finished
   end
+
+  def self.calculate_win_ratio(won, finished)
```

Comment:
This is a fairly effective efficiency hack. Everywhere we call `user.win_ratio`, we're also calling `user.games_won` and `user.games_lost`, so calling `games_won` and `games_lost` in `win_ratio` as well, both of which query the database, seems wasteful.

I ran a benchmark getting the rankings for a data set with 100 players and 1,000 games, and it consistently takes about half the time by using the cached variables. It's the difference between .45 seconds and .21 to load the full list of 100 on my macbook (which is beastly compared to an AWS micro instance). This is significant on users#index, and `.stats_string` sees modest speed gains too. Once that page starts taking a quarter second to load though we might want to consider removing win/loss/ratio from the page or storing it somewhere.